File: create-icons.js
const fs = require('fs');
const { createCanvas } = require('canvas');

const sizes = [72, 96, 128, 144, 152, 192, 384, 512];

function drawIcon(ctx, size) {
    const s = size / 512;

    // Background gradient
    const bgGrad = ctx.createLinearGradient(0, 0, size, size);
    bgGrad.addColorStop(0, '#0a0a2e');
    bgGrad.addColorStop(1, '#1a1a4a');

    // Rounded rectangle background
    const radius = 100 * s;
    ctx.beginPath();
    ctx.moveTo(radius, 0);
    ctx.lineTo(size - radius, 0);
    ctx.quadraticCurveTo(size, 0, size, radius);
    ctx.lineTo(size, size - radius);
    ctx.quadraticCurveTo(size, size, size - radius, size);
    ctx.lineTo(radius, size);
    ctx.quadraticCurveTo(0, size, 0, size - radius);
    ctx.lineTo(0, radius);
    ctx.quadraticCurveTo(0, 0, radius, 0);
    ctx.closePath();
    ctx.fillStyle = bgGrad;
    ctx.fill();

    // Stars
    const stars = [
        [80, 100, 2], [150, 60, 1.5], [420, 80, 2], [380, 140, 1.5],
        [60, 200, 1], [450, 250, 2], [100, 400, 1.5], [400, 420, 2],
        [200, 450, 1], [320, 470, 1.5]
    ];
    ctx.fillStyle = 'rgba(255, 255, 255, 0.7)';
    stars.forEach(([x, y, r]) => {
        ctx.beginPath();
        ctx.arc(x * s, y * s, r * s, 0, Math.PI * 2);
        ctx.fill();
    });

    // Galaxy center
    ctx.save();
    ctx.translate(256 * s, 256 * s);
    ctx.rotate(-30 * Math.PI / 180);

    // Outer glow
    const glowGrad = ctx.createRadialGradient(0, 0, 0, 0, 0, 140 * s);
    glowGrad.addColorStop(0, 'rgba(138, 79, 255, 0.4)');
    glowGrad.addColorStop(0.5, 'rgba(74, 111, 255, 0.2)');
    glowGrad.addColorStop(1, 'rgba(74, 111, 255, 0)');

    ctx.fillStyle = glowGrad;
    ctx.beginPath();
    ctx.ellipse(0, 0, 140 * s, 80 * s, 0, 0, Math.PI * 2);
    ctx.fill();

    // Spiral arms
    ctx.strokeStyle = 'rgba(122, 159, 255, 0.6)';
    ctx.lineWidth = 8 * s;
    ctx.lineCap = 'round';

    ctx.beginPath();
    ctx.moveTo(0, 0);
    ctx.quadraticCurveTo(60 * s, -40 * s, 100 * s, -20 * s);
    ctx.quadraticCurveTo(140 * s, 10 * s, 120 * s, 60 * s);
    ctx.stroke();

    ctx.beginPath();
    ctx.moveTo(0, 0);
    ctx.quadraticCurveTo(-60 * s, 40 * s, -100 * s, 20 * s);
    ctx.quadraticCurveTo(-140 * s, -10 * s, -120 * s, -60 * s);
    ctx.stroke();

    // Core glow
    const coreGrad = ctx.createRadialGradient(0, 0, 0, 0, 0, 40 * s);
    coreGrad.addColorStop(0, 'rgba(255, 255, 238, 1)');
    coreGrad.addColorStop(0.5, 'rgba(255, 255, 255, 0.9)');
    coreGrad.addColorStop(1, 'rgba(200, 200, 255, 0.3)');

    ctx.fillStyle = coreGrad;
    ctx.beginPath();
    ctx.ellipse(0, 0, 40 * s, 25 * s, 0, 0, Math.PI * 2);
    ctx.fill();

    ctx.restore();

    // Telescope at bottom
    ctx.save();
    ctx.translate(256 * s, 420 * s);
    ctx.strokeStyle = 'rgba(102, 136, 204, 0.6)';
    ctx.lineWidth = 4 * s;

    ctx.beginPath();
    ctx.arc(0, 0, 25 * s, 0, Math.PI * 2);
    ctx.stroke();

    ctx.beginPath();
    ctx.moveTo(0, 25 * s);
    ctx.lineTo(0, 50 * s);
    ctx.stroke();

    ctx.beginPath();
    ctx.moveTo(-15 * s, 50 * s);
    ctx.lineTo(15 * s, 50 * s);
    ctx.stroke();

    ctx.restore();
}

// Check if canvas module is available
try {
    sizes.forEach(size => {
        const canvas = createCanvas(size, size);
        const ctx = canvas.getContext('2d');
        drawIcon(ctx, size);
        const buffer = canvas.toBuffer('image/png');
        fs.writeFileSync(`icons/icon-${size}.png`, buffer);
        console.log(`Created icon-${size}.png`);
    });
    console.log('All icons created successfully!');
} catch (e) {
    console.log('Canvas module not available. Please create icons manually using generate-icons.html');
    console.log('Or install canvas: npm install canvas');
}

File: create_icons.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw
import math
import os

sizes = [72, 96, 128, 144, 152, 192, 384, 512]

def draw_icon(size):
    # Create image with alpha channel
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    s = size / 512  # scale factor

    # Background - rounded rectangle
    radius = int(100 * s)
    bg_color = (15, 15, 45, 255)  # Dark blue

    # Draw rounded rectangle background
    draw.rounded_rectangle([(0, 0), (size-1, size-1)], radius=radius, fill=bg_color)

    # Stars
    stars = [
        (80, 100, 2), (150, 60, 1.5), (420, 80, 2), (380, 140, 1.5),
        (60, 200, 1), (450, 250, 2), (100, 400, 1.5), (400, 420, 2),
        (200, 450, 1), (320, 470, 1.5)
    ]
    for x, y, r in stars:
        sx, sy, sr = int(x * s), int(y * s), max(1, int(r * s))
        draw.ellipse([(sx-sr, sy-sr), (sx+sr, sy+sr)], fill=(255, 255, 255, 180))

    # Galaxy core position
    cx, cy = int(256 * s), int(256 * s)

    # Draw galaxy glow (multiple ellipses for gradient effect)
    for i in range(5, 0, -1):
        rx, ry = int(140 * s * i / 5), int(80 * s * i / 5)
        alpha = int(40 + (5 - i) * 20)
        color = (100, 80, 200, alpha)
        # Approximate ellipse rotation by drawing at offset
        draw.ellipse([(cx-rx, cy-ry), (cx+rx, cy+ry)], fill=color)

    # Draw spiral arms (simplified as arcs)
    arm_color = (122, 159, 255, 150)
    line_width = max(2, int(6 * s))

    # Draw some curved lines to suggest spiral arms
    points1 = []
    points2 = []
    for t in range(20):
        angle = t * 0.15
        r1 = 20 * s + t * 5 * s
        r2 = 20 * s + t * 5 * s
        x1 = cx + int(r1 * math.cos(angle - 0.5))
        y1 = cy + int(r1 * 0.5 * math.sin(angle - 0.5))
        x2 = cx - int(r2 * math.cos(angle - 0.5))
        y2 = cy - int(r2 * 0.5 * math.sin(angle - 0.5))
        points1.append((x1, y1))
        points2.append((x2, y2))

    if len(points1) > 1:
        draw.line(points1, fill=arm_color, width=line_width)
        draw.line(points2, fill=arm_color, width=line_width)

    # Galaxy core (bright center)
    core_r = int(35 * s)
    for i in range(5, 0, -1):
        r = int(core_r * i / 5)
        alpha = int(100 + (5 - i) * 30)
        color = (255, 255, 240, min(255, alpha + 50))
        draw.ellipse([(cx-r, cy-int(r*0.6)), (cx+r, cy+int(r*0.6))], fill=color)

    # Bright core center
    inner_r = int(15 * s)
    draw.ellipse([(cx-inner_r, cy-int(inner_r*0.6)), (cx+inner_r, cy+int(inner_r*0.6))],
                 fill=(255, 255, 250, 255))

    # Telescope at bottom
    tx, ty = int(256 * s), int(420 * s)
    tel_color = (102, 136, 204, 150)
    tel_r = int(25 * s)
    line_w = max(2, int(3 * s))

    # Telescope circle (lens)
    draw.ellipse([(tx-tel_r, ty-tel_r), (tx+tel_r, ty+tel_r)], outline=tel_color, width=line_w)

    # Telescope body
    draw.line([(tx, ty+tel_r), (tx, ty+int(50*s))], fill=tel_color, width=line_w)

    # Telescope base
    draw.line([(tx-int(15*s), ty+int(50*s)), (tx+int(15*s), ty+int(50*s))],
              fill=tel_color, width=line_w)

    return img

# Create icons directory if it doesn't exist
os.makedirs('icons', exist_ok=True)

# Generate all sizes
for size in sizes:
    img = draw_icon(size)
    filename = f'icons/icon-{size}.png'
    img.save(filename, 'PNG')
    print(f'Created {filename}')

print('All icons created successfully!')
